chore(rollup): coinbase | prover rewards not limited to sender

Maddiaa0 opened the pull request:
## Overview

Allow coinbase rewards to be claimable by anyone, but they always go to the coinbase
Enables support for split contracts as rewards

Maddiaa0 commented:
* **#16280** <a href="https://app.graphite.dev/github/pr/AztecProtocol/aztec-packages/16280?utm_source=stack-comment-icon" target="_blank"><img src="https://static.graphite.dev/graphite-32x32-black.png" alt="Graphite" width="10px" height="10px"/></a> 👈 <a href="https://app.graphite.dev/github/pr/AztecProtocol/aztec-packages/16280?utm_source=stack-comment-view-in-graphite" target="_blank">(View in Graphite)</a>
* `next`


<h2></h2>

This stack of pull requests is managed by <a href="https://graphite.dev?utm-source=stack-comment"><b>Graphite</b></a>. Learn more about <a href="https://stacking.dev/?utm_source=stack-comment">stacking</a>.
<!-- Current dependencies on/for this PR: -->

just-mitch reviewed (approved):
Looks good, but some nits:
- I feel that `_sequencer` is a better arg name to `claimSequencerRewards`
- similarly `_prover` for claim prover rewards
- would be good to show in the tests that Alice can claim for bob

Maddiaa0 commented:
@claude do your thing

Maddiaa0 commented:
> @claude do your thing

damn i really thought that would do something

Maddiaa0 committed:
chore(rollup): coinbase | prover rewards not limited to sender

## Overview

Allow coinbase rewards to be claimable by anyone, but they always go to the coinbase
Enables support for split contracts as rewards
